repose-aggregator/components/filters/keystone-v2/src/main/scala/org/openrepose/filters/keystonev2/KeystoneV2Filter.scala

```diff
@@ -645,6 +669,58 @@ class KeystoneV2Filter @Inject()(configurationService: ConfigurationService,
     override def isInitialized: Boolean = initialized
   }
 
+  object CacheInvalidationFeedListener extends AtomFeedListener {
+    private var registeredFeeds = List.empty[String]
+
+    def invalidationEnabled(): Boolean = registeredFeeds.nonEmpty
+
+    def unRegisterFeeds(): Unit = {
+      registeredFeeds.synchronized {
+        registeredFeeds.foreach(feed =>
+          atomFeedService.unregisterListener(feed)
+        )
+      }
+    }
+
+    def registerFeeds(feeds: List[AtomFeedType]): Unit = {
+      registeredFeeds.synchronized {
+        unRegisterFeeds()
```

Comment:
We shouldn't just drop and re add here, your are going to potentially lose events which means people aren't necessarily going to get evicted like they should and that's a problem.